src/pybind/mgr/dashboard/services/auth.py

```diff
@@ -1,13 +1,101 @@
 # -*- coding: utf-8 -*-
 from __future__ import absolute_import
 
+from base64 import b64encode
+from datetime import datetime, timedelta
+import json
+import os
+import threading
 import time
+import uuid
 
 import cherrypy
+import jwt
 
-from .access_control import LocalAuthenticator
+from .access_control import LocalAuthenticator, UserDoesNotExist
 from .. import mgr, logger
-from ..tools import Session
+
+
+class JwtManager(object):
+    JWT_TOKEN_BLACKLIST_KEY = "jwt-token-black-list"
+    JWT_TOKEN_TTL = 28800  # default 8 hours
+    JWT_ALGORITHM = 'HS256'
+    _secret = None
+
+    LOCAL_USER = threading.local()
+
+    @staticmethod
+    def _gen_secret():
+        secret = os.urandom(16)
+        return b64encode(secret).decode('utf-8')
+
+    @classmethod
+    def init(cls):
+        # generate a new secret if it does not exist
+        secret = mgr.get_store('jwt_secret', None)
+        if secret is None:
+            secret = cls._gen_secret()
+            mgr.set_store('jwt_secret', secret)
+        cls._secret = secret
+
+    @classmethod
+    def gen_token(cls, username):
+        ttl = mgr.get_config('jwt-token-ttl', cls.JWT_TOKEN_TTL)
```

Comment:
s/jwt-token-ttl/jwt_token_ttl

I think config keys should always use underscores, see ``ceph config-key ls``.